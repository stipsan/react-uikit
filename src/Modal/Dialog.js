import classNames from 'classnames'
import { Component, PropTypes, createElement } from 'react'

export default class Dialog extends Component {
  static defaultProps = {
    blank: false,
    caption: '',
    close: true,
    footer: [],
    header: '',
    large: false,
    lightbox: false,
  }

  static propTypes = {
    children: PropTypes.node.isRequired,
    handleClose: PropTypes.func.isRequired,
    blank: PropTypes.bool,
    caption: PropTypes.string,
    close: PropTypes.bool,
    footer: PropTypes.array, // eslint-disable-line react/forbid-prop-types
    header: PropTypes.string,
    large: PropTypes.bool,
    lightbox: PropTypes.bool,
  }

  handleOverlayClick = event => event.stopPropagation()

  handleClose = (callback) => {
    this.props.handleClose(callback)
  }

  render() {
    const { handleOverlayClick, handleClose } = this
    const {
      children,
      header,
      footer,
      caption,
      lightbox,
      blank,
      close,
      large,
     } = this.props

    const dialogClassName = classNames('uk-modal-dialog', {
      'uk-modal-dialog-lightbox': lightbox,
      'uk-modal-dialog-blank': blank,
      'uk-modal-dialog-large': large,
    })
    const closeClassName = classNames('uk-modal-close uk-close', {
      'uk-close-alt': lightbox,
    })

    return (
      <div className={dialogClassName} onClick={handleOverlayClick}>
        {close && (
          <a
            className={closeClassName}
            onClick={handleClose}
          />
        )}
        {header && (<div className="uk-modal-header"><h2>{header}</h2></div>)}
        {caption && (<div className="uk-modal-caption">{caption}</div>)}
        {children}
        { footer.length > 0 &&
          (<div className="uk-modal-footer uk-text-right">
            {
              footer.map((component, i) => createElement(component, { key: i, handleClose }))
            }
          </div>)
        }
      </div>
    )
  }
}
