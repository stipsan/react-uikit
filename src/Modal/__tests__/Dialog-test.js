import renderer from 'react-test-renderer'
import Dialog from '../Dialog'

describe('Dialog', () => {
  const component = renderer.create(
    <Dialog handleClose={jest.fn()} isOpen={false} >Lorem Ipsum</Dialog>
  )
  it('renders correctly', () => {
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should handle handleClose', () => {
    const instance = component.getInstance()
    instance.handleClose()
    expect(instance.props.handleClose).toHaveBeenCalled()
  })

  it('should handle handleOverlayClick', () => {
    const instance = component.getInstance()
    const event = { stopPropagation: jest.fn() }
    instance.handleOverlayClick(event)
    expect(event.stopPropagation).toHaveBeenCalled()
  })
})
