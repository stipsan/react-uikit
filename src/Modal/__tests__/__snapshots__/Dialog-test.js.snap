// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog renders correctly 1`] = `
<div
  className="uk-modal-dialog"
  onClick={[Function]}
>
  <a
    className="uk-modal-close uk-close"
    onClick={[Function]}
  />
  
  
  Lorem Ipsum
</div>
`;
